### PR TITLE
iscsi: REST API ssl support

### PR DESCRIPTION
**Pull request by rjfd**
This PR adds the options to enable/disable SSL in ceph-iscsi REST API.
Currently SSL is disabled by default due to a bug in the communication between ceph-dashboard and ceph-iscsi when SSL is enabled.

Signed-off-by: Ricardo Dias <rdias@suse.com>

### CODE REVIEW

File: srv/salt/ceph/igw/config/create_iscsi_config.sls [hunk 1]
```diff
@@ -14,7 +14,40 @@
         trusted_ip_list: {{ ip_addresses }}
 {% endfor %}
 
+{% if pillar.get('ceph_iscsi_ssl', False) %}
+{% if pillar.get('ceph_iscsi_ssl_cert', None) and pillar.get('ceph_iscsi_ssl_key', None) %}
+
+/srv/salt/ceph/igw/cache/tls/certs/iscsi-gateway.crt:
+  file.managed:
+    - source: {{ pillar['ceph_iscsi_ssl_cert'] }}
+    - template: jinja
```
Comment (jan--f) [inline srv/salt/ceph/igw/config/create_iscsi_config.sls]: This probably shouldn't run through a jinja interpreter? Same lin line 33

Comment (rjfd) [reply srv/salt/ceph/igw/config/create_iscsi_config.sls]: you're right, copy&paste leftover. Will change it.

Comment (rjfd) [reply srv/salt/ceph/igw/config/create_iscsi_config.sls]: sorry for the delay, change is done.

**Comment by rjfd** [comment]
@susebot run teuthology

**Pull request merged**